src/upload/index.ts

```diff
@@ -49,6 +50,9 @@ export const upload = (
       else if(Array.isArray(param)) {
         formData.append(i, JSON.stringify(param));
       }
+      else if(typeof param === "object" && !(param instanceof File || param instanceof Blob)) {
```

Comment:
Node.js sdk is doing it in a better way. You can replace this whole thing.
https://github.com/imagekit-developer/imagekit-nodejs/blob/master/libs/upload/index.ts#L50